AK/String.cpp

```diff
@@ -182,6 +182,17 @@ template Optional<u32> String::to_uint(TrimWhitespace) const;
 template Optional<unsigned long> String::to_uint(TrimWhitespace) const;
 template Optional<unsigned long long> String::to_uint(TrimWhitespace) const;
 
+#ifndef KERNEL
+template<typename T>
+Optional<T> String::to_double(TrimWhitespace trim_whitespace) const
+{
+    return StringUtils::convert_to_double<T>(*this, trim_whitespace);
+}
+
+template Optional<double> String::to_double(TrimWhitespace) const;
+template Optional<float> String::to_double(TrimWhitespace) const;
```

Comment:
Shouldn't this be called `to_float()`? Or something like `to_floading_point()`, `too_double<float>()` seems kinda weird :sweat_smile:

Reply:
Yeah I went back and forth on this, since the other functions are called `to_int` and `to_uint` and this might make it easier to find. But I'll change it since I agree it is confusing

Reply:
Ended up splitting it to `to_double` and `to_float`